### DISCUSSION TITLE
Cache on iOS

### DISCUSSION THREAD

**Discussion by ComplexCarbos** [jhomlala/betterplayer#904] [Q&A]
Is anyone running into  problems enabling the cache on iOS? It was working fine for me up until a few weeks ago. Now when I enable it, I get these errors in the log:

`PINDiskCache.m (1223) ERROR: The item couldn’t be saved because the file name “aHR0cHM6Ly92aWRlb2RlbGl2ZXJ5Lm5ldC85YWYzMTc4NDQ4OWQ0ZjA4OWE3NzQwMDk2MDRhN2M1Zi92aWRlby8xMDgwL3NlZ18xMi50cz9wPWV5SjBlWEJsSWpvaWRISmhibk50ZFhnaUxDSjJhV1JsYjBsRUlqb2lPV0ZtTXpFM09EUTBPRGxrTkdZd09EbGhOemMwTURBNU5qQTBZVGRqTldZaUxDSnZkMjVsY2tsRUlqb3hOVFE0TnpVNU1Dd2ljMlZuYldWdWRFUjFjbUYwYVc5dVUyVmpjeUk2TkN3aWRISmhZMnNpT2lKalpXTTNZemRqWW1Jd01UQTBORGhpWWpSbE9UTTRObVV4T1RZeVlqRTJOeUlzSW5KbGJtUnBkR2x2YmlJNklqazJOamMzT1RJeklpd2liWFY0YVc1bklqb2lNVFEzTWpFME56RTBJbjAmcz1LTUtrTFY3Q3BzS2Z3cjNEc1dUQ2hzT2xUTU9wVmNLeVBjS0J3NHJDaGNPU3dwNFl3cTBkRjhLb0MxTjd3b0hEbDI0” is invalid.`

As far as I can tell, the error is generated due to a cache key that is too large?

If I disable cache, these errors go away. Works fine on Android.

I believe I'm using current/updated tools:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.0, on macOS 12.1 21C52 darwin-x64, locale en-US)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] VS Code (version 1.49.1)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

```

I'm using the most recent version of better_player (0.0.81)

**Comment by ndahlquist** [comment]
I believe we ran into the same issue when using Cloudflare Stream as an HLS host: they use long URLs.

The issue appears to be caused by the HLSCache dependency. We made a patch here: https://github.com/SwarmMedia/HLSCachingReverseProxyServer/commit/75896b113d42e249e3a94d7df552e0bd7f1c6d39.

**Comment by ComplexCarbos** [comment]
Hmm, my videos are hosted via Cloudflare too. 

What is interesting tho, is that I've always used Cloudflare and the caching did work in the past.
